scripts/core/editor3/reducers/editor3.jsx

```diff
@@ -54,6 +54,26 @@ export const forceUpdate = (state) => {
     };
 };
 
+
+function noCharactersAreSelectedAndCursorIsInsideCommentOrAnnotation(editorState) {
+    var selection = editorState.getSelection();
+    var character = editorState
+        .getCurrentContent()
+        .getBlockForKey(selection.getAnchorKey())
+        .getCharacterList()
+        .get(selection.getAnchorOffset());
+
+    if (character === undefined) {
+        return false;
+    }
+
+    var characterStyle = character.getStyle();
+
+    return selection.getAnchorKey() === selection.getFocusKey()
+        && selection.getAnchorOffset() === selection.getFocusOffset()
+        && (characterStyle.contains('COMMENT') || characterStyle.contains('ANNOTATION'));
```

Comment:
here it could be [character.hasStyle](https://draftjs.org/docs/api-reference-character-metadata.html#hasstyle) so we stick with draftjs api